Content.Shared/Medical/Cryogenics/InsideCryoPodComponent.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Robust.Shared.GameStates;
+
+namespace Content.Shared.Medical.Cryogenics;
+
+[RegisterComponent]
+[NetworkedComponent]
+public sealed class InsideCryoPodComponent: Component
+{
+    [ViewVariables]
+    public EntityUid Holder;
+
+    public Vector2 PreviousOffset { get; set; } = new(0, 0);
+}
```

Comment:
Doesn't look like this handles serialization, same with the container code in cryopodsystem, so if someone saves a map with something inside a cryo pod it will break.

Reply:
this still being worked on? shame if we dont eventually see this finished.

Reply:
Yeah I am, I have an exam soon and I need to sort that out first. After that, I need to figure out how to move the interactions in shared as suggested (some of them like drag&drop use doafters and I remember that causes problems since they aren't predicted yet?). I also need to figure out what exactly needs to be done to handle serialization (simply datafield it? I need to understand how serialization works at all here before continuing). And the only issue left after this is the ParentColdDamageThreshold override, i'm really not sure how to handle recursive parent changes. I need to see if there's already something else that does this (most likely yes).
I expect to be able to work on this again on this weekend at worst, sorry for the wait!

Reply:
I've added datafields for those two properties, but I don't understand how exactly I should handle containers for the cryo pod. As far as I saw, other instances have a NonSerializable decorator on ContainerSlot fields. Other than that I didn't see any custom logic that handles serialization/deserialization, could you give me an example?